src/Uno.Extensions.Hosting.UI/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace Uno.Extensions.Hosting;
+
+/// <summary>
+/// Extensions for the <see cref="IApplicationBuilder" />.
+/// </summary>
+public static class ApplicationBuilderExtensions
+{
+	/// <summary>
+	/// Creates a new Shell for the main content of the Window, builds the <see cref="IHost" />
+	/// and starts the Host before finally returning it.
+	/// </summary>
+	/// <typeparam name="TShell">The <see cref="UIElement" /> to set as the <see cref="Window" /> content.</typeparam>
+	/// <param name="builder">The <see cref="IApplicationBuilder" />.</param>
+	/// <returns>The started <see cref="IHost" />.</returns>
+	public static async Task<IHost> ShowAsync<TShell>(this IApplicationBuilder builder)
```

Comment:
This shouldn't exist in Hosting, well not in its current form. It's unlikely that this would ever be used as is. Most common (other than using navigation) would be to create frame and nav to first page, before activating window.

On this topic, we need to review where Activate is called in the showasync and navigateasync methods to ensure nav is complete before Activate is called.

Reply:
removed...